tinygrad/ops.py

```diff
@@ -162,8 +162,8 @@ def assert_valid(op:LazyOp, st:ShapeTracker):
       st = op.arg.st if op.op in BufferOps else sts[op.src[0]]
       for x in op.src:
         if sts[x].shape != st.shape:
-          if prod(sts[x].shape) == prod(st.shape): raise AssertionError(f"found implicit reshape {x.op} {op.op} {sts[x].shape} != {sts[x].shape}")
-          raise AssertionError(f"found implicit expand {x.op} {sts[x].shape} != {op.op} {sts[x].shape} {prod(sts[x].shape)} != {prod(st.shape)}")
+          if prod(sts[x].shape) == prod(st.shape): raise AssertionError(f"found implicit reshape {x.op} {op.op} {sts[x].shape} != {st.shape}")
```

Comment:
If you want to do a PR with this bug fix I'd merge

Reply:
Sure, moved into separate PR